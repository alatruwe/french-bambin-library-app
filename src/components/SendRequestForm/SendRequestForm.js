import React, { Component } from "react";
import ValidationError from "../ValidationError/ValidationError.js";

export default class SendRequest extends Component {
  constructor(props) {
    super(props);
    this.state = {
      object: {
        value: "",
        touched: false,
      },
      message: {
        value: "",
        touched: false,
      },
    };
  }
  static defaultProps = {
    history: {
      push: () => {},
    },
  };

  updateObject(object) {
    this.setState({ object: { value: object, touched: true } });
  }
  updateMessage(message) {
    this.setState({
      message: {
        value: message,
        touched: true,
      },
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.history.push(`/home`);
  }
  validateObject() {
    const object = this.state.object.value.trim();
    if (object.length === 0) {
      return "Email object is required";
    }
  }
  validateMessage() {
    const message = this.state.message.value.trim();
    if (message.length === 0) {
      return "A message is required";
    }
  }

  render() {
    const objectError = this.validateObject();
    const messageError = this.validateMessage();

    return (
      <form onSubmit={(e) => this.handleSubmit(e)}>
        <div className="form-group">
          <label htmlFor="text">Object</label>
          <input
            type="text"
            name="object"
            id="object"
            placeholder="I'm interested in your item"
            required
            onChange={(e) => this.updateObject(e.target.value)}
          />
          {this.state.object.touched && (
            <ValidationError message={objectError} />
          )}
        </div>
        <div className="form-group">
          <label htmlFor="text">Message</label>
          <textarea
            type="text"
            name="message"
            id="message"
            onChange={(e) => this.updateMessage(e.target.value)}
          />
          {this.state.message.touched && (
            <ValidationError message={messageError} />
          )}
        </div>

        <div>
          <button
            type="submit"
            disabled={this.validateObject() || this.validateMessage()}
          >
            Send
          </button>
        </div>
      </form>
    );
  }
}
