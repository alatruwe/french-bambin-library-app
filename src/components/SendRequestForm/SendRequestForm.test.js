import React from "react";
import ReactDOM from "react-dom";
import SendRequestForm from "./SendRequestForm";
import renderer from "react-test-renderer";

describe(`SendRequestForm component`, () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<SendRequestForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(<SendRequestForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
