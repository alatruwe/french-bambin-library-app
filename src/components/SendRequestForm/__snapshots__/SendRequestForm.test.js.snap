// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendRequestForm component renders the UI as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      htmlFor="text"
    >
      Object
    </label>
    <input
      id="object"
      name="object"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="text"
    >
      Message
    </label>
    <input
      id="message"
      name="message"
      onChange={[Function]}
      type="text"
    />
  </div>
  <div>
    <button
      disabled="Email object is required"
      type="submit"
    >
      Send
    </button>
  </div>
</form>
`;
